javalib/src/main/scala/java/util/package.scala

```diff
@@ -29,7 +31,196 @@ package object util {
   private[util] def defaultOrdering[E]: Ordering[E] = {
     new Ordering[E] {
       def compare(a: E, b: E): Int =
-        a.asInstanceOf[Comparable[E]].compareTo(b)
+        if (a === null) {
+          if (b === null) 0
+          else -1
+        } else if (b === null) {
+          1
+        } else {
+          a.asInstanceOf[Comparable[E]].compareTo(b)
+        }
```

Comment:
So this change should be reverted to just `a.asInstanceOf[Comparable[E]].compareTo(b)` as it was before, with the change to `add()` that I suggested.